engine/tests/service/test_orchestrator38.py

```
@@ -35,6 +38,19 @@ def save(self, value) -> None:
         super().__setitem__(key, value)
 
 
+class RoundTripPersistence(dict):
+    """Persistence shim that stores ledgers as unstructured payloads."""
+
+    def save(self, value) -> None:
+        key = getattr(value, "uid", None)
+        if key is None:
+            raise ValueError("Unable to determine key for saved value")
```

Ruff (0.15.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)